### ISSUE TITLE
npm: Prisma CLI can only execute he first command you try on Windows (in cmd/PowerShell)

### ISSUE THREAD

**Issue by janpio** [denoland/deno#16255]
To reproduce:
1. Install deno, make sure `deno` works in `cmd` or Powershell
2. Run a `prisma` command via Deno, .e.g. `deno run -A --unstable npm:prisma@4.4.0-integration-feat-deno-deploy.1`
3. Run another Prisma CLI command via Deno, e.g. `deno run -A --unstable npm:prisma@4.4.0-integration-feat-deno-deploy.1 init` 
4. Observe error message: 
   ```
   Error: Can't write to C:\Users\Jan\AppData\Local\deno\npm\registry.npmjs.org\@prisma\engines\4.4.0-integration-feat-deno-deploy.1 please make sure you install "prisma" with the right permissions.
   ```

You can then reset the state by deleting the global deno npm folder (for me `C:\Users\Jan\AppData\Local\deno\npm\registry.npmjs.org`). When you then run the second command directly, it succeeds. If you run the first command afterwards though, it will also fail with the same error message.

This does not happen when using the deno CLI via WSL/Ubuntu.

---

```
PS C:\Users\Jan\Documents\throwaway\denoTest> deno --version
deno 1.26.1 (release, x86_64-pc-windows-msvc)
v8 10.7.193.3
typescript 4.8.3
```

Full output:
```
PS C:\Users\Jan\Documents\throwaway\denoTest> deno run -A --unstable npm:prisma@4.4.0-integration-feat-deno-deploy.1

Prisma is a modern DB toolkit to query, migrate and model your database (https://prisma.io)

Usage

  $ prisma [command]

Commands

            init   Set up Prisma for your app
        generate   Generate artifacts (e.g. Prisma Client)
              db   Manage your database schema and lifecycle
         migrate   Migrate your database
          studio   Browse your data with Prisma Studio
          format   Format your schema

Flags

     --preview-feature   Run Preview Prisma commands

Examples

  Set up a new Prisma project
  $ prisma init

  Generate artifacts (e.g. Prisma Client)
  $ prisma generate

  Browse your data
  $ prisma studio

  Create migrations from your Prisma schema, apply them to the database, generate artifacts (e.g. Prisma Client)
  $ prisma migrate dev

  Pull the schema from an existing database, updating the Prisma schema
  $ prisma db pull

  Push the Prisma schema state to the database
  $ prisma db push

PS C:\Users\Jan\Documents\throwaway\denoTest> deno run -A --unstable npm:prisma@4.4.0-integration-feat-deno-deploy.1 init
Error: Can't write to C:\Users\Jan\AppData\Local\deno\npm\registry.npmjs.org\@prisma\engines\4.4.0-integration-feat-deno-deploy.1 please make sure you install "prisma" with the right permissions.
```

**Comment by janpio** [comment]
Ok, this error message is actually from Prisma itself, or the package `@prisma/engines` which is mentioned in the error message:
https://github.com/prisma/prisma/blob/3391f771e7e3aac3c4befb3545ab16c9c42395a2/packages/fetch-engine/src/download.ts#L423-L432

Here is more output with some debug output:
```
PS C:\Users\Jan\Documents\throwaway\denoTest> deno run -A --unstable npm:prisma@4.4.0-integration-feat-deno-deploy.1 init
  prisma:engines  binaries to download libquery-engine, migration-engine, introspection-engine, prisma-fmt +0ms
  prisma:download  file C:\Users\Jan\AppData\Local\deno\npm\registry.npmjs.org\@prisma\engines\4.4.0-integration-feat-deno-deploy.1\query_engine-windows.dll.node does not exist and must be downloaded +0ms
  prisma:download  file C:\Users\Jan\AppData\Local\deno\npm\registry.npmjs.org\@prisma\engines\4.4.0-integration-feat-deno-deploy.1\migration-engine-windows.exe does not exist and must be downloaded +1ms
  prisma:download  file C:\Users\Jan\AppData\Local\deno\npm\registry.npmjs.org\@prisma\engines\4.4.0-integration-feat-deno-deploy.1\introspection-engine-windows.exe does not exist and must be downloaded +0ms
  prisma:download  file C:\Users\Jan\AppData\Local\deno\npm\registry.npmjs.org\@prisma\engines\4.4.0-integration-feat-deno-deploy.1\prisma-fmt-windows.exe does not exist and must be downloaded +0ms
Error: Error: Can't write to C:\Users\Jan\AppData\Local\deno\npm\registry.npmjs.org\@prisma\engines\4.4.0-integration-feat-deno-deploy.1 please make sure you install "prisma" with the right permissions.
    at downloadBinary (file:///C:/Users/Jan/AppData/Local/deno/npm/registry.npmjs.org/@prisma/engines/4.4.0-integration-feat-deno-deploy.1/dist/index.js:23150:13)
    at file:///C:/Users/Jan/AppData/Local/deno/npm/registry.npmjs.org/@prisma/engines/4.4.0-integration-feat-deno-deploy.1/dist/index.js:22955:18
    at Array.map (<anonymous>)
    at download (file:///C:/Users/Jan/AppData/Local/deno/npm/registry.npmjs.org/@prisma/engines/4.4.0-integration-feat-deno-deploy.1/dist/index.js:22954:26)
    at async ensureBinariesExist (file:///C:/Users/Jan/AppData/Local/deno/npm/registry.npmjs.org/@prisma/engines/4.4.0-integration-feat-deno-deploy.1/dist/index.js:23262:3)
    at async _CLI.parse (file:///C:/Users/Jan/AppData/Local/deno/npm/registry.npmjs.org/prisma/4.4.0-integration-feat-deno-deploy.1/build/index.js:96090:9)
    at async main (file:///C:/Users/Jan/AppData/Local/deno/npm/registry.npmjs.org/prisma/4.4.0-integration-feat-deno-deploy.1/build/index.js:98251:18)
```

So Prisma CLI executes this `download` script (via its bundle) at some point, and decides that it needs to download 4 files. Before it does that for the first file (and debug logs the filename) it checks if it can a) access that folder and b) create the folder (via using https://www.npmjs.com/package/make-dir) - and that throws and triggers this error message.

Unfortunately the actual error is not logged or output, so it not super clear if a) or b) are triggering that error, and what the actual error is.

Side note: The folder `C:\Users\Jan\AppData\Local\deno\npm\registry.npmjs.org\@prisma\engines\4.4.0-integration-feat-deno-deploy.1` actually exists already and also has content. Via git bash:
```
Jan@Surface-Studio MINGW64 ~/AppData/Local/deno/npm/registry.npmjs.org/@prisma/engines
$ ls -l
total 736
drwxr-xr-x 1 Jan 197609      0 Oct 12 19:31 4.4.0-integration-feat-deno-deploy.1/
-rw-r--r-- 1 Jan 197609 749425 Oct 12 19:31 registry.json
```

**Comment by kt3k** [comment]
@janpio Thank you for the report and analysis. This looks like the problem of how we polyfill `fs.accessSync` in `deno_std/node` https://github.com/denoland/deno_std/blob/deef340b5557f65c5f4de52f5a66a68496db2bb0/node/_fs/_fs_access.ts#L75 (We're currently ignoring the case when the file is only available to the owner and it's in windows. We need to somehow polyfill that case on windows.)

**Comment by janpio** [comment]
Any idea when this will be released? Somehow my workaround is not working any more, and I am now getting the error message every time :(

**Comment by bartlomieju** [comment]
@janpio in a coupe hours, I'm cutting the release right now.

**Comment by janpio** [comment]
Can confirm this is fixed in `v1.26.2` 🥳
